.github/workflows/typos.yml

```diff
@@ -0,0 +1,41 @@
+# Copyright 2022 Datafuse Labs
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: TyposCheck
+
+on: [push, pull_request]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-${{ github.event_name }}
+  cancel-in-progress: true
+
+env:
+  RUST_BACKTRACE: 1
+
+jobs:
+  typos-check:
+    name: typos check
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    env:
+      FORCE_COLOR: 1
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          clean: "true"
+
+      - run: cargo install typos-cli --locked
```

Comment:
How about using https://github.com/taiki-e/install-action for better cache support

Reply:
ok,but

> If a tool not included in the list above is specified, this action uses [cargo-binstall](https://github.com/cargo-bins/cargo-binstall) as a fallback.

```
Calling `cargo-binstall` to do the install
 INFO resolve: Resolving package: 'typos-cli'
 WARN The package typos-cli v1.13.12 will be installed from source (with cargo)
```